app/models/cart.rb

```diff
@@ -32,20 +32,28 @@ def get_items # rubocop:disable AccessorMethodName
   end
 
   # Adds equipment model id to items hash
-  def add_item(equipment_model)
+  def add_item(equipment_model, _quantity = nil)
     return if equipment_model.nil?
     key = equipment_model.id
     items[key] = items[key] ? items[key] + 1 : 1
   end
 
   # Remove equipment model id from items hash, or decrement its count
-  def remove_item(equipment_model)
+  def remove_item(equipment_model, _quantity = nil)
     return if equipment_model.nil?
     key = equipment_model.id
     items[key] = items[key] ? items[key] - 1 : 0
     self.items = items.except(key) if items[key] <= 0
   end
 
+  def edit_cart_item(equipment_model, quantity)
+    return if equipment_model.nil?
+    return if quantity < 0
+    key = equipment_model.id
+    items[key] = items[key] ? quantity : 0
+    self.items = items.except(key) if items[key] <= 0
```

Comment:
Do you need to test for `items[key] <= 0` or could you just test `items[key] == 0`? You'd have `return`ed already if the quantity was less than zero.